File: Java/Inheritance/InterfaceTest2/src/Worker.java
package taxation;

public class Worker implements TaxPayer {

	private double monthlyWages;

	public Worker(double wages) { monthlyWages = wages; }

	public double annualIncome() {
		double bonus = monthlyWages < 20000 ? 30000 : 1.5 * monthlyWages;
		return 12 * monthlyWages + bonus;
	}
}

File: DotNet/Inheritance/InterfaceTest2/DemoApp/Taxation/Worker.cs
namespace Taxation;

public struct Worker : ITaxPayer
{
    public decimal MonthlyWages;

    public decimal AnnualIncome()
    {
        decimal bonus = MonthlyWages < 20000 ? 30000 : 1.5M * MonthlyWages;
        return 12 * MonthlyWages + bonus;
    }
}